### PR TITLE
Further clean up after PR 1719 for Nano Every

### PR DESCRIPTION
**Pull request by ngyl88**
Changes:
1. Complete removal of `ARDUINO_AVR_NANO_EVERY` conditionals from `fastpin_avr.h` to avoid unintentional overwriting pin definition. Arduino Nano Every is subset of ATmega4809 .
2. Extract file `fastpin_avr_atmega4809.h` from `fastpin_avr.h`

More details on:
- https://github.com/FastLED/FastLED/pull/1719#issuecomment-2359697145

### CODE REVIEW

**Comment by ngyl88** [comment]
@zackees could you help to review this ?

BTW are you able to configure the reviewer list for this repo? We don't see any reviewer after creating the PR.

**Comment by zackees** [comment]
I do not have the ability. But you are free to @ me so that I see it.

Please squash your commits so that there is only one commit in the future. It's extremely hard to follow a CL when there are X number of commits to it.

**Comment by ngyl88** [comment]
Okay, I will squash all commits into one in the future. Usually, I grouped my commit through squash and rebase, eg. `d0423f2`. I leave it as separate commits when context are separate, so that it's easier to trace the history when merged into the `master` branch.

All changes regardless of commits could be seen at https://github.com/FastLED/FastLED/pull/1720/files .
